src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.MovieDao;
+import com.dev.cinema.exceptions.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.Movie;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
+            return movie;
```

Comment:
As I understood from checklist, here we need to user try with finally with session and don't set Id, cz Hibernate will do it

Reply:
Done